spyder/plugins/editor/widgets/tests/conftest.py

```diff
@@ -99,7 +100,14 @@ def teardown():
     return editor, fallback
 
 
-@pytest.fixture(scope="module")
+# Windows tests fail if using module scope
+if os.name == 'nt':
+    LSP_PLUGIN_SCOPE = 'module'
+else:
+    LSP_PLUGIN_SCOPE = 'function'
```

Comment:
I don't like this. It'll make our tests way slower because it'll create a new LSP server per function.

Reply:
>I don't like this. It'll make our tests way slower because it'll create a new LSP server per function.

It fails consistently locally and on CI for me and tests do not become as slow as you think.

The gains of using a fixture per module are really visible on the main windows tests, no ton the LSP tests.

Reply:
But it was not failing before. So why do you need to do this change?

Reply:
>But it was not failing before. So why do you need to do this change?

It was, just intermittently. With the changes I made now it fails consistently so the "random" error is now reproducible.